liquibase-core/src/main/java/liquibase/database/core/EnterpriseDBDatabase.java

```diff
@@ -1,16 +1,40 @@
 package liquibase.database.core;
 
+import liquibase.Scope;
 import liquibase.database.DatabaseConnection;
+import liquibase.database.jvm.JdbcConnection;
 import liquibase.exception.DatabaseException;
 import liquibase.statement.SqlStatement;
 import liquibase.statement.core.RawSqlStatement;
+import liquibase.util.JdbcUtil;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
 
 public class EnterpriseDBDatabase extends PostgresDatabase {
 
     @Override
     public boolean isCorrectDatabaseImplementation(DatabaseConnection conn) throws DatabaseException {
         final String url = conn.getURL();
-        return url.contains("edb") || (url.contains("postgres") && url.contains(":5444"));
+        if (url.startsWith("jdbc:edb:") || url.startsWith("jdbc:postgres")) {
```

Comment:
Will this work with diffs and snapshots where the url potentially uses offline?  like the following **offline:<db_type>?snapshot**
In this case, it would be offline:edb?snapshot

Reply:
In those cases, the "offline:edb" part forces it to be edb so it doesn't need to use this method.